lib/configurators/aws/aws.go

```diff
@@ -942,6 +938,26 @@ func buildSecretsManagerStatements(fileConfig *config.FileConfig, target awslib.
 	return statements
 }
 
+// buildSTSAssumeRoleStatements returns AWS IAM statements necessary for
+// assuming AWS IAM roles.
+func buildSTSAssumeRoleStatements(flags configurators.BootstrapFlags, targetCfg targetConfig, boundary bool) ([]*awslib.Statement, error) {
+	if len(targetCfg.assumesAWSRoles) == 0 {
+		return nil, nil
+	}
+	if boundary {
+		return []*awslib.Statement{{
+			Effect:    awslib.EffectAllow,
+			Actions:   stsActions,
+			Resources: []string{"*"},
```

Comment:
wildcard in boundary policy so it doesn't implicitly deny assuming a role that trusts the identity directly.
In permissions policy, we don't use `*` because then the role will have more permissions than it needs.